product_variant_default_code/views/product_view.xml

```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0"?>
+<odoo>
+
+    <record id="product_template_only_form_view_inh_variantdefaultcode"
+            model="ir.ui.view">
+        <field name="name">
+            product.template.only.form.view.inh.variantdefaultcode
+        </field>
+        <field name="model">product.template</field>
+        <field name="inherit_id"
+               ref="product.product_template_only_form_view"/>
+        <field name="arch" type="xml">
+            <field name="default_code" position="after">
+                <field name="prefix_code"
+                       attrs="{'invisible':
+                          [('attribute_line_ids', '=', [])]}"/>
+                <field name="reference_mask"
+                       attrs="{'invisible':
+                          [('attribute_line_ids', '=', [])]}"
+                       groups="product_variant_default_code.group_product_default_code"/>
+            </field>
+        </field>
+    </record>
+
+    <record id="product_normal_form_view_inh_variantdefaultcode"
+            model="ir.ui.view">
+        <field name="name">
+            product.normal.form.view.inh.variantdefaultcode
+        </field>
+        <field name="model">product.product</field>
+        <field name="inherit_id" ref="product.product_normal_form_view"/>
+        <field name="arch" type="xml">
+            <field name="default_code" position="before">
+                <field name="manual_code" />
+            </field>
+        </field>
+    </record>
+
```

Comment:
You also should add this xml : 
```xml
    <record id="product_variant_easy_edit_view" model="ir.ui.view">
        <field name="model">product.product</field>
        <field name="inherit_id" ref="product.product_variant_easy_edit_view"/>
        <field name="arch" type="xml">
            <field name="default_code" position="before">
                <field name="manual_code" />
            </field>
        </field>
    </record>
```
to allow displaying this field in alternate view